fvtest/compilerunittest/CodeGenTest.hpp

```diff
@@ -36,6 +40,115 @@ class CodeGenTest : public TRTest::CompilerUnitTest {
     TR::Node *fakeNode;
 };
 
+class BinaryInstruction {
+
+public:
+
+    uint8_t _buf[INSTRUCTION_MAX_LENGTH] = {0};
+    size_t _size = 0;
+
+    BinaryInstruction() {}
+
+    BinaryInstruction(uint32_t instr) {
+        append(instr);
+    }
+
+    BinaryInstruction(uint32_t a, uint32_t b) {
+        append(a);
+        append(b);
+    }
+
+    BinaryInstruction(uint8_t *instr, size_t size) : _size(size) {
+        TR_ASSERT_FATAL(size <= INSTRUCTION_MAX_LENGTH, "unexpected instruction length");
+        memcpy(_buf, instr, size);
+    }
+
+    BinaryInstruction(const char *instr) : BinaryInstruction((uint8_t *) instr, strlen(instr)) {}
```

Comment:
I figure there can be an extra constructor for this purpose, but none of the tests have needed it at this point. I don't think it makes sense to change all the other tests to include their length.

Reply:
I'm going to have to disagree with you here. I think leaving this constructor around at all is just asking for trouble. Nothing would prevent somebody from accidentally including a 0 byte in their expected result or warn them that this has occurred. This could lead to a lot of confusion and wasted time in the future that I'd rather avoid.

Perhaps a better solution would be to use a template constructor that takes in a reference to an array with a templated size, e.g.

```cxx
template <size_t size>
BinaryInstruction(const char (&instr)[size]) : BinaryInstruction(reinterpret_cast<const uint8_t*>(instr), size - 1) {}
```

I'm not 100% sure this will work with all our compilers (AIX in particular has historically been problematic when it comes to some uses of templates) so this probably needs to be tested to see if it works with our CI, but if it does it should allow string literals to be passed in with their size always determined correctly AFAIK.

Reply:
I think parsing should also work fine